plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/DefaultThriftMetastoreClientFactory.java

```diff
@@ -97,14 +117,56 @@ public DefaultThriftMetastoreClientFactory(
                 Optional.ofNullable(config.getSocksProxy()),
                 config.getMetastoreTimeout(),
                 metastoreAuthentication,
-                nodeManager.getCurrentNode().getHost());
+                nodeManager.getCurrentNode().getHost(),
+                buildThriftHttpContext(config));
     }
 
     @Override
     public ThriftMetastoreClient create(HostAndPort address, Optional<String> delegationToken)
             throws TTransportException
     {
-        return create(() -> createTransport(address, delegationToken), hostname);
+        return create(() -> getTransportSupplier(address, delegationToken), hostname);
+    }
+
+    private TTransport getTransportSupplier(HostAndPort address, Optional<String> delegationToken)
+            throws TTransportException
+    {
+        if (thriftHttpContext.isPresent()) {
+            return createHttpTransport(address, thriftHttpContext.get());
+        }
+        else {
+            return createTransport(address, delegationToken);
+        }
+    }
+
+    public TTransport createHttpTransport(HostAndPort address, ThriftHttpContext httpThriftContext)
+            throws TTransportException
+    {
+        HttpClientBuilder clientBuilder = createHttpClientBuilder(address, httpThriftContext);
+        URI uri = URI.create(String.format("%s://%s:%s%s",
```

Comment:
You could do
```java
new URI(context.getScheme(), null, address.getHost(), address.getPort(), context.getHttpPath(), null, null);
```
This seems better than building a string and then parsing it

Reply:
Done